pandas/core/indexes/datetimelike.py

```diff
@@ -589,7 +589,11 @@ def delete(self, loc):
 
     @doc(NDArrayBackedExtensionIndex.insert)
     def insert(self, loc: int, item):
-        result = super().insert(loc, item)
+        try:
+            result = super().insert(loc, item)
+        except (ValueError, TypeError):
+            # i.e. self._data._validate_scalar raised
```

Comment:
really? it seems that the base can handle this no?

Reply:
in the non-raising case we still need to do the _get_insert_freq a few lines down